data/moves.js

```diff
@@ -17355,14 +17355,20 @@ let BattleMovedex = {
 		accuracy: 100,
 		basePower: 80,
 		category: "Special",
-		desc: "Has a higher chance for a critical hit.",
+		desc: "Has a higher chance for a critical hit. Ignores redirection from moves and abilities.",
 		shortDesc: "High critical hit ratio.",
 		id: "snipeshot",
 		name: "Snipe Shot",
 		pp: 15,
 		priority: 0,
 		flags: {protect: 1, mirror: 1},
 		critRatio: 2,
+		effect: {
+			onRedirectTargetPriority: 3,
+			onRedirectTarget(target, source, source2, move) {
+				return target;
+			},
+		},
```

Comment:
Don't you need to add the volatile? (Also remove it again, in case this was executed due to Instruct.)

Reply:
(yes)

Reply:
That, or try a different approach. I like a move flag saying "ignore redirects".

Reply:
I'm confused. What volatile are you talking about?

Reply:
When you add an `effect` block, you're creating an effect, but you have to add it as a volatile (or status, weather, etc.) for it to actually do anything. Because you never add the effect, it doesn't have any... effect... (sorry I couldn't think of a way out of the pun)

Reply:
This is all irrelevant, because this should be a move-property-boolean rather than a custom effect.